### PR TITLE
save and restore the chrome session local storage information

### PR DESCRIPTION
**Pull request by devoncarew**
## Description

This PR saves and restores the chrome session local storage information for flutter web chrome devices. When running chrome devices, we already save and restore the userDataDir preferences file, containing things like the window size and location. This PR also saves are restores the local storage info for web apps, allowing the local storage info to persist across app runs.

## Related Issues

None, but this addresses an issue found while developing DevTools. I'll link the PR in here once that work is complete.

## Tests

I updated the following tests:
- `flutter_tools/test/general.shard/web/chrome_test.dart`: `'can seed chrome temp directory with existing preferences'` (renamed to `'can seed chrome temp directory with existing session data'`)

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_tools/lib/src/web/chrome.dart [hunk 1]
```diff
@@ -206,6 +189,54 @@ class ChromeLauncher {
...
+    final File targetPreferencesFile = _fileSystem.file(_fileSystem.path.join(cacheDir?.path ?? '', _preferencesPath));
+    final File sourcePreferencesFile = _fileSystem.file(_fileSystem.path.join(userDataDir.path, _preferencesPath));
+    final Directory targetLocalStorageDir = _fileSystem.directory(_fileSystem.path.join(cacheDir?.path ?? '', _localStoragePath));
+    final Directory sourceLocalStorageDir = _fileSystem.directory(_fileSystem.path.join(userDataDir.path, _localStoragePath));
+
+    if (sourcePreferencesFile.existsSync()) {
+      targetPreferencesFile.parent.createSync(recursive: true);
+      // If the file contains a crash string, remove it to hide the popup on next run.
+      final String contents = sourcePreferencesFile.readAsStringSync();
+      targetPreferencesFile.writeAsStringSync(contents
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/web/chrome.dart]: Is there a link you can add in a comment either describing the schema of this file, or at least of this field?

Also, is there a test for the find/replace?

File: packages/flutter_tools/lib/src/web/chrome.dart [hunk 2]
```diff
@@ -115,35 +116,25 @@ class ChromeLauncher {
...
-    // Seed data dir with previous state.
-
-    final File savedPreferencesFile = _fileSystem.file(_fileSystem.path.join(dataDir?.path ?? '', preferencesPath));
-    final File destinationFile = _fileSystem.file(_fileSystem.path.join(activeDataDir.path, preferencesPath));
-    if (dataDir != null) {
-      if (savedPreferencesFile.existsSync()) {
-        destinationFile.parent.createSync(recursive: true);
-        savedPreferencesFile.copySync(destinationFile.path);
-      }
+    final Directory userDataDir = _fileSystem.systemTempDirectory.createTempSync('flutter_tool.');
```
[the remark this answers is not in the document]
Comment (devoncarew) [reply packages/flutter_tools/lib/src/web/chrome.dart]: Happy yo change the name (note however that this is the name used by the existing code).

File: packages/flutter_tools/lib/src/web/chrome.dart (hunk 1 above)
Comment (devoncarew) [reply packages/flutter_tools/lib/src/web/chrome.dart]: I don't believe there's a schema. The existing text (this portion of the PR is existing code) does contain a check for the find / replace for the crash string.

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/web/chrome.dart]: I found one of file locations by reading chrome source, we could link to that?

Comment (devoncarew) [reply packages/flutter_tools/lib/src/web/chrome.dart]: Can you link it here, and I'll add to the PR?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/web/chrome.dart]: I can't find it, It might have just been grepping the user data dir for strings

Comment (devoncarew) [reply packages/flutter_tools/lib/src/web/chrome.dart]: I found some info here: https://www.chromium.org/developers/design-documents/preferences; will in-line into the source code.

**Comment by devoncarew** [comment]
ptal - updated based on review comments

**Review by zanderso — APPROVED** [review]
lgtm

**Pull request merged**